server/src/main/java/org/opensearch/search/startree/StarTreeQueryContext.java

```diff
@@ -184,6 +189,31 @@ private static boolean validateKeywordTermsAggregationSupport(
         return true;
     }
 
+    private static boolean validateRangeAggregationSupport(
+        CompositeDataCubeFieldType compositeIndexFieldInfo,
+        AggregatorFactory aggregatorFactory
+    ) {
+        if (!(aggregatorFactory instanceof RangeAggregatorFactory rangeAggregatorFactory)) {
+            return false;
+        }
+
+        // Validate request field is part of dimensions
+        if (compositeIndexFieldInfo.getDimensions()
```

Comment:
will this work for date dimension ? if not we might need numeric dimension type check or a better mechanism for validation

Reply:
Thanks for pointing this out. This does not works for date dimensions. 
For now, I am adding a type check for this and a TODO to support date type in future.

I'll follow up to add support for date type once I add up utils to support date type as part of https://github.com/opensearch-project/OpenSearch/issues/17443